vm/sdk/multisig/tx.go

```diff
@@ -0,0 +1,152 @@
+package multisig
+
+import (
+	"bytes"
+	"fmt"
+	"maps"
+	"slices"
+
+	// FIXME: use go-scale when we add a tag to encode uint8 non-compact.
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	athcon "github.com/athenavm/athena/ffi/athcon/bindings/go"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/vm/core"
+	"github.com/spacemeshos/go-spacemesh/vm/sdk"
+)
+
+// SpawnArguments contains a collection with PublicKeys.
+type SpawnArguments struct {
+	Required   uint8
+	PublicKeys []core.PublicKey
+}
+
+type SpendArguments struct {
+	To     types.Address
+	Amount uint64
+}
+
+// Signatures is a collections of parts that must satisfy multisig
+// threshold requirement.
+type Signatures []Part
+
+// Part contains a reference to public key and signature from private key counterpart.
+type Part struct {
+	Ref uint8
+	Sig core.Signature
+}
+
+func NewAggregator(unsigned []byte) *Aggregator {
+	return &Aggregator{unsigned: unsigned, parts: map[uint8]Part{}}
+}
+
+// Aggregator is a signature accumulator.
+type Aggregator struct {
+	unsigned []byte
+	parts    map[uint8]Part
+}
+
+// Add signature parts to the accumulator.
+func (tx *Aggregator) Add(ref uint8, sig core.Signature) {
```

Comment:
`s/tx/ag/` or something similar.
What do you think about renaming `Aggregator` to `SignatureAggregator` to be more precise? or about moving Aggregator into subpackage to use ns for inducing more precision naming?

That's only suggestions, not a blocker in general for PoC.